Tests/NetworkServiceTests/URLParametersEncoderTests/QueryItemsTests.swift

```diff
@@ -0,0 +1,68 @@
+import NetworkService
+import XCTest
+
+final class QueryItemsTests: XCTestCase {
+    func testReturnsQueryItemWithExpectedNameAndValue() {
+        // Given
+        let urlParameters = ["a": "a"]
+        let expectedQueryItems = [URLQueryItem(name: "a", value: "a")]
+
+        // When
+        let actualQueryItems = URLParametersEncoder.queryItems(from: urlParameters)
+
+        // Then
+        XCTAssertEqual(actualQueryItems, expectedQueryItems)
+    }
+
+    func testReturnsEmptyArrayWhenPassedEmptyDictionaryOfURLParameter() {
+        // Given // When
+        let actualQueryItems = URLParametersEncoder.queryItems(from: URLParameters())
+        // Then
+        XCTAssertEqual(actualQueryItems, [])
+    }
+
+    func testReturnsQueryItemWithExpectedNameAndValueWhenPassedURLParameterHasEmptyKeyAndValue() {
+        // Given
+        let urlParameters = ["": ""]
+        let expectedQueryItems = [URLQueryItem(name: "", value: "")]
+
+        // When
+        let actualQueryItems = URLParametersEncoder.queryItems(from: urlParameters)
+
+        // Then
+        XCTAssertEqual(actualQueryItems, expectedQueryItems)
+    }
+
+    func testReturnsQueryItemWithExpectedNameAndValueWhenPassedURLParameterHasSpaceInKeyAndInValue() {
+        // Given
+        let urlParameters = [" ": " "]
+        let expectedQueryItems = [URLQueryItem(name: " ", value: " ")]
+
+        // When
+        let actualQueryItems = URLParametersEncoder.queryItems(from: urlParameters)
+
+        // Then
+        XCTAssertEqual(actualQueryItems, expectedQueryItems)
+    }
+
+    func testReturnsAmountOfQueryItemsThatEqualsToAmountOfURLParameters() throws {
```

Comment:
Why do we need tests for 2 and 3 parameters, it doesn't look like an edge case.
I can understand if you would add cases with 5 or 10 parameters, but 2 and 3 look excessive to me.

I prefer the separate test cases for 2 and 3, 5 or 10 parameters.

Reply:
What do 5 and 10 cases give? I wrote cases with 2 and 3 elements to show 2 different values. The case with one element seemed to me not representative enough.
What edge cases do you suggest? 🙂
For 5 and 10 cases, I'll need to create 2 dictionaries with such number of elements. And what will we get for it?